test/unit/constraints/string.test.ts

```diff
@@ -436,11 +432,10 @@ describe("String constraints", function () {
             const input = "test";
             const object = new MyClass(input);
             const result = validator.validate(object);
-            assert.property(result, "error");
-            assert.isNull(result.error, "Validation should pass");
-            assert.property(result, "value");
-            const value = result.value;
-            assert.equal(object.myProperty, input);
+            expect(result).toHaveProperty('error');
+            expect(result.error).toBeNull();
```

Comment:
Note that in a couple places some assertions will not have the same friendly failure message they previously had like on this line. From what I gathered, jest does not have this feature, but that said, I don't think it's terribly needed with jest's easy to read BDD assertion style and when there is a failure jest tells you the exact line along with a pretty well formatted expectation and actual result, so it's not something I've really missed.

Reply:
🙏 That's a shame. I'd like to keep the friendly messages in some way, it cuts down dev time when debugging tests.

Suggested options:
1. Make the friendly messages TS comments.
1. Add functions named similarly to the friendly messages, move the assertions into them. (Names could be quite verbose.)
1. Look at adding something like [jest-expect-message](https://github.com/mattphillips/jest-expect-message). (The types will shortly land in [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/34890)).

(3 is my preference)

Reply:
> cuts down dev time when debugging tests.

Okay, keep in mind that the call to `it` already provides a great explanation for what the test is checking for, which gets displayed when the test fails, along with the exact failing line.

Anyway, I've added a custom matcher to jest now so many tests have been simplified to `expect(instance).toBeValid()` or `expect(instance).not.toBeValid()`. 👍